client/js/models/archive.js

```diff
@@ -191,7 +191,7 @@ module.exports = {
           if (file && file.progressListener && file.progressHandler) {
             file.progressListener.removeListener('progress', file.progressHandler)
           }
-          if (file.fullPath === '/dat.json') send('archive:updateMetadata', {}, noop)
+          if (file.fullPath === '/dat.json' || file.fullPath === '/datapackage.json') send('archive:updateMetadata', {}, noop)
```

Comment:
shouldn't this be handled by the `archive.list()` stream? because it's also possible that this file is added via a remote peer.